be/src/formats/json/numeric_column.cpp

```diff
@@ -39,10 +39,30 @@ static Status add_column_with_numeric_value(FixedLengthColumn<T>* column, const
     }
 
     case simdjson::ondemand::number_type::unsigned_integer: {
-        uint64_t in = value->get_uint64();
         T out{};
+        // When get_uint64 throws an exception here, we need to recheck it with get_int64.
+        // This is because currently there exists a bug in simdjson, for number <= -9223372036854775808,
+        // simdjson will recognized it as unsigned_integer, and use get_uint64() to parse it, which will
+        // raise an exception here. So here we need to recheck by get_int64().
+        // This bug in simdjson is expected to be solved by the following commit, lator we can remove
+        // the redundant error handling logic here.
+        // https://github.com/simdjson/simdjson/commit/b169dc2ea71bc8e9296de9749c42d78f80c8a633
+        bool checked_cast_flag = true;
+        uint64_t in = 0;
+        try {
+            in = value->get_uint64();
+            checked_cast_flag = checked_cast(in, &out);
+        } catch (simdjson::simdjson_error& e) {
+            // only needs to handle int64_t::min here, throw exception for any other number
+            // get_int64() itself may also throw exception
+            if (value->get_int64() == std::numeric_limits<int64_t>::min()) {
+                checked_cast_flag = checked_cast(std::numeric_limits<int64_t>::min(), &out);
+            } else {
+                throw simdjson::simdjson_error(simdjson::error_code::NUMBER_OUT_OF_RANGE);
```

Comment:
Is it better to return an error instead?

Reply:
In the original code, it should raise an exception here because it can not be parsed by simdjson get_uint64(). 
Here we recheck if it is a std::numeric_limits<int64_t>::min() due to a bug in simdjson. If it is not, i think we should still throw an exception here?